Updated several components to use AP LO's; Corrected exercise types used in iReading (SP/Pers)/HW (builder/SP/Pers)/PW

Dantemss opened the pull request:
Will be ready to be merged after https://github.com/openstax/tutor-server/pull/439 is merged.

jpslav commented:
@joemsak you should just fold this into your work and then @Dantemss can close this out.

Dantemss commented:
Rebased onto the no_units branch (so this branch now has no bio units)
@cnuber This fixes the bio learning guide (you will have to force pull it since I rebased onto another branch)

Dantemss commented:
@jpslav I'd say merge Joe's PR first, then we can rebase onto it.

joemsak commented:
It should be possible for me to either merge this onto, or rebase my course guide work onto this branch, doesnt look like we changed the same files. Let me know what you guys want

jpslav commented:
Will merge joes pr first. 


> On Jul 10, 2015, at 10:28 AM, Joe Sak <notifications@github.com> wrote:
> 
> It should be possible for me to either merge this onto, or rebase my course guide work onto this branch, doesnt look like we changed the same files. Let me know what you guys want
> 
> —
> Reply to this email directly or view it on GitHub.
>

Dantemss commented:
Need to unrebase this from my branch and rebase onto JP's...

jpslav commented:
@Dantemss ping me when this one is ready.

Dantemss commented:
@jpslav ready
https://docs.google.com/spreadsheets/d/1rNmlvtgM7eGON17IpaB-lA-rltJlIeIVbRmgly8-T-4/edit#gid=369864926

Dantemss commented:
This PR is required for the bio learning guide to work.

jpslav commented on app/controllers/api/v1/courses_controller.rb:
```diff
@@ -75,7 +75,13 @@ def exercises
+    review_exercises = Content::Routines::SearchExercises[tag: 'ost-chapter-review']
+    review_exercise_ids = review_exercises.pluck(:id)
+    review_exercise_relation = Content::Models::Exercise.where(id: review_exercise_ids)
+    search_outputs = SearchLocalExercises.call(relation: review_exercise_relation,
```
The double call is because we don't do boolean tag queries?

Dantemss replied:
Yeah

jpslav commented on app/routines/get_course_guide.rb:
```diff
@@ -129,8 +134,9 @@ def compile_pages(task_steps)
+    page_aplos = page_data.aplos
     task_steps.select do |task_step|
-      (task_step.tasked.los & page_los).any?
+      (task_step.tasked.los & page_los).any? || (task_step.tasked.aplos & page_aplos).any?
```
@Dantemss all this aplo stuff is totally what we have to do right now, but as I imagine you agree in the fall we have got to figure out something different, otherwise when we add another book with some other kind of LO, we're going to have to add yet another blahlo.  /cc @kevinburleigh75

Dantemss replied:
I'm hoping there are no more blahlo's but sure...
